lib/haxe/src/org/apache/thrift/transport/THttpClient.hx

```diff
@@ -39,12 +46,25 @@ class THttpClient extends TTransport {
     private var requestBuffer_  : BytesOutput = new BytesOutput();
     private var responseBuffer_ : BytesInput = null;
 
+    #if !js
     private var request_        : Http = null;
+    #else
+    private var request_        : JsHttp = null;
+    #end
 
 
     public function new( requestUrl : String) : Void {
-          request_ = new Http(requestUrl);
-        request_.addHeader( "contentType", "application/x-thrift");
+        #if !js
+        request_ = new Http(requestUrl);
+        #else
+        request_ = new JsHttp(requestUrl);
+        request_.async = false;
+        #end
+        request_.addHeader("Content-Type", "application/x-thrift");
+        request_.addHeader("Accept", "application/x-thrift");
```

Comment:
this will need redesign of api as well, because transport doesn't have access to protocol type information